### ISSUE TITLE
Inconsistencies when instantiation wxColour and comparison to None (and maybe others too)

### ISSUE THREAD

**Issue by jensgoe** [wxWidgets/Phoenix#600]
**Operating system**: Win7 64bit
**wxPython version**: (4, 0, 0, 'b2')
**Stock or custom build**: stock
**Python version**: 2.7.13 (64bit)
**Stock or custom build**: stock

**Description of the problem**:

In Classic Python there are a lot of helper methods for a lot of objects. Colour, Font, Point, etc. I didn't check every object but see the sample below for wxColour.

```python
# Classic wxPython (3.0.2)
>>> col = wx.Colour()
>>> col == None
False
>>> col
wx.Colour(0, 0, 0, 255)

# wxPython 4.0.0 b2
>>> col = wx.Colour()
>>> col == None
True
>>> col
wx.Colour(-1, -1, -1, 255)
```

**Comment by jensgoe** [comment]
Currently i  have no problem with it. I just noticed it. I know for comparing with None pep8 suggests using "... is None".

**Comment by RobinD42** [comment]
This one is intentional. 

`wx.Colour()` is the same as `wx.NullColour` which indicates that the colour is uninitialized, or invalid, or similar, like how NULL would be used for a pointer value in C++ to indicate the same kinds of things. Since `None` in Python is used for the same kinds of things then the code that auto-converts Python values to a `wx.Colour` object explicitly checks for `None`, and converts it to a `wx.NullColour`.

So

```
someColour == None
```
 first converts the `None` to a `wx.NullColour` and then passes it to `wx.Colour.__eq__`

```
>>> wx.Colour()
wx.Colour(-1, -1, -1, 255)
>>> wx.NullColour
wx.Colour(-1, -1, -1, 255)
>>>
>>> wx.NullColour == wx.Colour()
True
>>> wx.NullColour == None
True
>>> wx.NullColour is None
False
```

So, in other words, `wx.NullColour` is equal to (or equivalent to) `None`, but it is not the `None` object.

**Comment by RobinD42** [comment]
For the curious: https://github.com/wxWidgets/Phoenix/blob/wxPython-4.0.0b2/etg/colour.py#L186

**Comment by RobinD42** [comment]
This issue has been mentioned on **Discuss wxPython**. There might be relevant details there:

https://discuss.wxpython.org/t/solved-how-to-remove-styling-in-textctrl/37126/9
